### PR TITLE
Add algorithm "DELLA" 

### PR DESCRIPTION
**Pull request by 6DammK9**
From [DELLA-Merging: Reducing Interference in Model Merging through Magnitude-Based Sampling](https://arxiv.org/abs/2406.11617)

Unit test passed (and updated). 
E2E Model is being examined. It requires a lot more time (4-6x) to merge for my scale (192 models).
Should be *comfortable* for < 20 models, which the complexity is still in control ( $O(n log n)$ which is [merge sort](https://en.wikipedia.org/wiki/Merge_sort) in the bottom layer ).

### CODE REVIEW

**Comment by 6DammK9** [comment]
~~As in Discord~~ Regarding to [this topic](https://discuss.pytorch.org/t/speedup-torch-argsort-torch-sort-runtime-and-sorting-axis-how-to/131441/2), `torch.argsort` can use GPU which is a lot faster even it is called for 2 times.
However, this implies detecting device which I'm not familiar in how to handle this case "in framework".
It is fine to leave the `rankdata` for both CPU and GPU mode (GPU pass to CPU for that part) because the IO / CPU overhead seems acceptable in practical merging scale (eeyep, no one merge 20+ models like me, which must use CPU mode), meanwhile making detection / making excessive "parameters" / testing the `stable` part of `argsort` sounds sketchy.

**Comment by ljleb** [comment]
> However, this implies detecting device which I'm not familiar in how to handle this case "in framework".

Hey no worries, I can take care of it if you want. It is quite straightforward to do when you know how: the only thing you need to add is `.to(device=a.device)` or something like that to the inputs to put them on the right device. You also need to pass `device=a.device` to any function that creates tensors like `torch.ones`, `torch.arange`, etc. I'm putting this here in case you want to do it but I will eventually get to it.

BTW apologies for the delay on this PR, it looks very good overall. I am really busy with life at the moment. I will try to find some time soon to test locally and then it should be good to go.
